examples/miscellaneous/parallel_backend_benchmarks.py

```diff
@@ -0,0 +1,197 @@
+"""
+=======================================================================
+Comparing parallelization with context manager using different backends
+=======================================================================
+
+In this notebook we demo how to use a context manager to change the default
+backend used by the joblib implementation inside any scikit-learn object that
+has a parameter `n_jobs`.
+
+In practice, whether increasing the number of workers is helpful at improving
+runtime depends on many factors. It is usually better to experiment rather than
+assuming that it is always a good thing. In some cases it can be highly
+detrimental to performance to run multiple copies of some estimators or
+functions in parallel (see :ref:`oversubscription`).
+
+To run this notebook locally, you may need to install dask and ray. These
+packages can be installed with `pip install dask-ml "ray[default]"`.
+
+For more information see the :ref:`User Guide <parallelism>`.
+
+"""
+
+# %%
+import sys
+import joblib
+
+try:
+    import dask
+except ImportError:
+    print("The package 'dask' is required to run this example.")
+    sys.exit()
+
+try:
+    import ray
+except ImportError:
+    print("The package 'ray' is required to run this example.")
+    sys.exit()
+
+print(joblib.__version__)
+print(dask.__version__)
+print(ray.__version__)
+
+# %%
+# We build a classification task using
+# :class:`~sklearn.datasets.make_classification` and cross-validate an
+# :class:`~sklearn.ensemble.HistGradientBoostingClassifier` with default
+# parameters on top of it.
+
+# %%
+from sklearn.datasets import make_classification
+from sklearn.ensemble import HistGradientBoostingClassifier
+
+X, y = make_classification(n_samples=1_000, random_state=0)
+hist_gbdt = HistGradientBoostingClassifier(random_state=0)
```

Comment:
Does `joblib` automatically `OMP_NUM_THREADS` for backends such as `dask` or `ray`?

Reply:
Does `joblib` control the number of OpenMP threads for other backends? There can be over-subscription because of dask parallelism + OpenMP parallelism.

XREF: I know for `loky`, joblib will control [the environment variables](https://github.com/joblib/joblib/blob/df122868c0a27cd4f47a1d8f8e398c0d3cc3c907/joblib/_parallel_backends.py#L154-L163), but I have not checked for other backends.

Reply:
That's a good question. I don't think so but I think we should investigate. It's possible that ray and dask do something smart by default outside of the of joblib connector itself.